lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -1267,6 +1267,10 @@ class ParseableInterfaceModuleLoaderImpl {
       depsAdjustedToMTime.push_back(adjustedDep);
     }
 
+    // Create the module cache if we haven't created it yet.
+    StringRef parentDir = path::parent_path(outputPath);
+    (void)llvm::sys::fs::create_directory(parentDir);
```

Comment:
I think it's better to use `create_directories`, and similarly for the existing call in `ParseableInterfaceModuleLoader.cpp`.

Reply:
if you pass `%t/nested/NonExistentCachePath` in the test case I believe it will fail if you don't use `create_directories`.